src/Separated.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * ```ts
+ * interface Separated<A, B> {
+ *    readonly left: A
+ *    readonly right: B
+ * }
+ * ```
+ *
+ * Represents a result of separating a whole into two parts.
+ *
+ * @since 2.10.0
+ */
+
+import { Functor2 } from './Functor'
+import { Bifunctor2 } from './Bifunctor'
+
+// -------------------------------------------------------------------------------------
+// model
+// -------------------------------------------------------------------------------------
+
+/**
+ * A `Separated` type which holds `left` and `right` parts.
+ *
+ * @since 2.0.0
+ */
+export interface Separated<A, B> {
+  readonly left: A
+  readonly right: B
+}
+
+// -------------------------------------------------------------------------------------
+// constructors
+// -------------------------------------------------------------------------------------
+
+/**
+ * @category constructors
+ * @since 2.10.0
+ */
+export const separated = <A, B>(left: A, right: B): Separated<A, B> => ({ left, right })
+
+// -------------------------------------------------------------------------------------
+// non-pipeables
+// -------------------------------------------------------------------------------------
+
+const map_ = <A, B, C>(fa: Separated<A, B>, f: (b: B) => C): Separated<A, C> => ({ ...fa, right: f(fa.right) })
+
+const mapLeft_ = <A, B, C>(fa: Separated<A, C>, f: (a: A) => B): Separated<B, C> => ({ ...fa, left: f(fa.left) })
+
+const bimap_ = <A, B, C, D>(fa: Separated<A, C>, g: (A: A) => B, f: (c: C) => D): Separated<B, D> => ({
+  left: g(fa.left),
+  right: f(fa.right)
+})
+
+// -------------------------------------------------------------------------------------
+// pipeables
+// -------------------------------------------------------------------------------------
+
+/**
+ * `map` can be used to turn functions `(a: A) => B` into functions `(fa: F<A>) => F<B>` whose argument and return types
+ * use the type constructor `F` to represent some computational context.
+ *
+ * @category Functor
+ * @since 2.10.0
+ */
+export const map = <B, C>(f: (b: B) => C) => <A>(fa: Separated<A, B>): Separated<A, C> => ({
```

Comment:
I'd use the `separated` constructor (everywhere), so `separated(fa.left, f(fa.right))`